src/main/java/org/fsg1/fmms/backend/database/Connection.java

```diff
@@ -37,71 +38,31 @@ public Connection(final Configuration config, final BasicDataSource connectionPo
      * Execute any query on the database using a <code>PreparedStatement</code>.
      *
      * @param columnName The name of the column that is returned by the query.
-     * @param query      The SQL String of the query you want to perform.
+     * @param statement  The SQL String of the query you want to perform.
      * @param parameters An optional array of Objects from which to fill the parameters.
      * @return A ResultSet of the query results.
      * @throws Exception if something goes wrong performing the query.
      */
-    public String executeQuery(final String columnName, final String query, final Object... parameters) throws Exception {
-        try (java.sql.Connection connection = startTransaction()) {
-            final String result = executeQuery(connection, columnName, query, parameters);
-            commitTransaction(connection);
-            return result;
-        }
-    }
+    public String executeQuery(final String columnName, final String statement, final Object... parameters) throws Exception {
+        try (java.sql.Connection connection = connectionPool.getConnection()) {
+            try (PreparedStatement preparedStatement = connection.prepareStatement(statement)) {
+                mapParams(preparedStatement, parameters);
 
-    /**
-     * Offers you a Connection from the connection pool and disables auto committing. Any queries executed on
-     * this connection will not go in effect until the commitTransaction method is called.
-     * Keep in mind the timeouts will be the driver's defaults.
-     * This connection will not be closed until the transaction is committed with the commitTransaction method.
-     *
-     * @return A connection to do transactions with.
-     * @throws SQLException If a database access error occurs, while participating in a distributed transaction
-     *                      or this method is called on a closed connection.
-     */
-    public java.sql.Connection startTransaction() throws SQLException {
-        final java.sql.Connection connection = connectionPool.getConnection();
-        connection.setAutoCommit(false);
-        return connection;
-    }
+                try (ResultSet result = preparedStatement.executeQuery()) {
+                    String emptyString = "";
+                    if (columnName == null) return emptyString;
```

Comment:
You "unnecessarily" always create an empty string. Whether it will be used or not.
Use: `return "";`

Reply:
I used to do that to clarify why I returned "". I didn't want to return null. But I did as you said